nifi-nar-bundles/nifi-druid-bundle/nifi-druid-controller-service/src/main/java/org/apache/nifi/controller/druid/DruidTranquilityController.java

```diff
@@ -231,10 +228,9 @@
     public static final PropertyDescriptor DIMENSIONS_LIST = new PropertyDescriptor.Builder()
             .name("druid-cs-dimensions-list")
             .displayName("Dimension Fields")
-            .description("A comma separated list of field names that will be stored as dimensions on ingest.")
-            .required(true)
-            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .description("A comma separated list of field names that will be stored as dimensions on ingest. Set to empty string for schema-less dimensions.")
```

Comment:
I believe you can keep the `NON_EMPTY_VALIDATOR` (instead of `VALID`) and just say in the doc to leave it blank if you want schema-less. If the property is not required, IIRC the framework doesn't bother calling the validator if it is not set. The way I've described will actually prevent you from explicitly setting it to the empty string, so you could use `NON_BLANK_VALIDATOR` if you still wanted to support that, but usually it's easier to say "don't set it unless you want it"